lib/components/AlbumScreen/preset_chip.dart

```diff
@@ -0,0 +1,153 @@
+    if (!_controller.hasClients) return false;
+    var offset =
+        (widget.chipWidth + 8.0) * widget.values.indexOf(currentValue) -
+            maxWidth / 2 +
+            widget.chipWidth / 2;
+
+    offset = min(
+        max(0, offset), maxWidth - _controller.position.maxScrollExtent + 15);
+
+    _controller.animateTo(
+      offset,
+      duration: Duration(milliseconds: 350),
+      curve: Curves.easeOutCubic,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    var nowActiveValue = widget.activeValue;
+
+    return LayoutBuilder(builder: (context, constraints) {
+      return SingleChildScrollView(
+        controller: _controller,
+        scrollDirection: Axis.horizontal,
+        child: Wrap(
+          spacing: 8.0,
+          runSpacing: 8.0,
+          crossAxisAlignment: WrapCrossAlignment.center,
+          children: List.generate(widget.values.length, (index) {
+            final currentValue = widget.values[index];
+            var newValue = "x$currentValue";
+
+            if (currentValue == nowActiveValue) {
+              if (scrolledAlready) {
+                scrollToActivePreset(currentValue, constraints.maxWidth);
+              } else {
+                Future.delayed(Duration(milliseconds: 200), () {
+                  scrollToActivePreset(currentValue, constraints.maxWidth);
+                });
+                scrolledAlready = true;
+              }
+            }
+
+            return PresetChip(
+              value: newValue,
+              backgroundColour: currentValue == nowActiveValue
+                  ? widget.mainColour?.withOpacity(0.4)
+                  : widget.mainColour?.withOpacity(0.1),
+              isTextBold: currentValue == 1.0,
+              width: widget.chipWidth,
+              onTap: () {
+                setState(() {
+                  nowActiveValue = currentValue;
+                });
+                _queueService.setPlaybackSpeed(currentValue);
+                widget.onPressed!();
+              },
+            );
+          }),
+        ),
+      );
+    });
+  }
+}
+
+class PresetChip extends StatelessWidget {
+  const PresetChip({
+    Key? key,
+    required this.width,
+    this.value = "",
+    this.onTap,
+    this.backgroundColour,
+    this.isTextBold,
+  }) : super(key: key);
+
+  final double width;
+  final String value;
+  final void Function()? onTap;
+  final Color? backgroundColour;
+  final bool? isTextBold;
+
+  @override
+  Widget build(BuildContext context) {
+    final backgroundColor = backgroundColour ?? _defaultBackgroundColour;
+    final color = Theme.of(context).textTheme.bodySmall?.color ?? Colors.white;
+
+    return SizedBox(
+      width: width,
+      height: _height,
+      child: Material(
+        color: backgroundColor,
+        borderRadius: _borderRadius,
+        child: InkWell(
+            onTap: onTap,
+            borderRadius: _borderRadius,
+            child: Center(
```

Comment:
I think this can be replaced with a simple `TextButton` with the appropriate `ButtonStyle`

Reply:
After trying this, it seems to make things more complicated. I would just keep it like it is right now.

Or is there a better way than doing the `TextButton` like this:
```
TextButton(
  style: ButtonStyle(
    backgroundColor: MaterialStateProperty.resolveWith<Color?>(
        (Set<MaterialState> states) {
      return backgroundColor;
    }),
  ),
  onPressed: onTap,
  child: Text(
    value,
    style: TextStyle(
      color: color,
      overflow: TextOverflow.visible,
      fontWeight: isTextBold! ? FontWeight.w700 : FontWeight.normal,
    ),
    softWrap: false,
  ),
),
```

Even with that, there is still a border radius one has to get rid of in the `ButtonStyle`.

Reply:
I changed that, it's not much shorter but now semantics are correct :)